pyquil/quil.py

```diff
@@ -275,10 +273,33 @@ def inst(self, *instructions: Union[InstructionDesignator, RSProgram]) -> "Progr
             elif isinstance(instruction, types.GeneratorType):
                 self.inst(*instruction)
             elif isinstance(instruction, tuple):
+                warnings.warn(
+                    "Adding instructions to a program by specifying them as tuples is deprecated. Consider building "
+                    "the instruction you need using classes from the `pyquil.gates` or `pyquil.quilbase` modules and "
+                    "passing those to Program.inst() instead.",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
                 if len(instruction) == 0:
                     raise ValueError("tuple should have at least one element")
+                elif len(instruction) == 1:
+                    self.inst(instruction[0])
                 else:
-                    self.inst(" ".join(map(str, instruction)))
+                    op = instruction[0]
+                    if op == "MEASURE":
+                        if len(instruction) == 2:
+                            self.measure(instruction[1], None)
+                        else:
+                            self.measure(instruction[1], instruction[2])
```

Comment:
Shouldn't this `else` check for `len == 3` and raise an exception if `len > 3`?

Reply:
Agreed, but this is how it was written in pyQuil 3, and since this code exists purely for backwards compatibility, I'm hesitant to add a new error case.